src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -0,0 +1,152 @@
+import slimeknights.tconstruct.library.utils.IdParser;
+import slimeknights.tconstruct.library.utils.Util;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public record MaxArmorStatModule(TinkerDataCapability.TinkerDataKey<Float> statKey, LevelingValue amount, TinkerDataCapability.ComputableDataKey<ModifierMaxLevel> maxLevel, boolean allowBroken, @Nullable TagKey<Item> heldTag, ArmorStatModule.TooltipStyle tooltipStyle, ModifierCondition<IToolStackView> condition) implements HookProvider, EquipmentChangeModifierHook, ModifierModule, TooltipModifierHook, ModifierCondition.ConditionalModule<IToolStackView> {
+  private static final List<ModuleHook<?>> TOOLTIP_HOOKS = HookProvider.<ArmorStatModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE, ModifierHooks.TOOLTIP);
+  private static final List<ModuleHook<?>> NO_TOOLTIP_HOOKS = HookProvider.<ArmorStatModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE);
+  public static final RecordLoadable<MaxArmorStatModule> LOADER = RecordLoadable.create(
+    new IdParser<>(ResourceLocation::new, "Stat").xmap((location, factory) -> TinkerDataCapability.TinkerDataKey.<Float>of(location), (tinkerDataKey, errorFactory) -> tinkerDataKey.getId()).requiredField("maxLevel", MaxArmorStatModule::statKey),
+    LevelingValue.LOADABLE.directField(MaxArmorStatModule::amount),
+    BooleanLoadable.INSTANCE.defaultField("allow_broken", false, MaxArmorStatModule::allowBroken),
+    Loadables.ITEM_TAG.nullableField("held_tag", MaxArmorStatModule::heldTag),
+    new EnumLoadable<>(ArmorStatModule.TooltipStyle.class).defaultField("tooltip_style", ArmorStatModule.TooltipStyle.NONE, MaxArmorStatModule::tooltipStyle),
+    ModifierCondition.TOOL_FIELD,
+    MaxArmorStatModule::new);
+
+  public MaxArmorStatModule(TinkerDataCapability.TinkerDataKey<Float> statKey, LevelingValue amount, boolean allowBroken, @Nullable TagKey<Item> heldTag, ArmorStatModule.TooltipStyle tooltipStyle, ModifierCondition<IToolStackView> condition) {
+    this(statKey, amount, TinkerDataCapability.ComputableDataKey.of(IdExtender.LocationExtender.INSTANCE.prefix(statKey.getId(), "_data"), ModifierMaxLevel::new), allowBroken, heldTag, tooltipStyle, condition);
+  }
+
+  @Override
+  public RecordLoadable<MaxArmorStatModule> getLoader() {
+    return LOADER;
+  }
+
+  @Override
+  public void addModules(ModuleHookMap.Builder builder) {
+    builder.addModule(new MaxArmorLevelModule(maxLevel, allowBroken, heldTag));
+  }
+
+  @Override
+  public List<ModuleHook<?>> getDefaultHooks() {
+    return tooltipStyle == ArmorStatModule.TooltipStyle.NONE ? NO_TOOLTIP_HOOKS : TOOLTIP_HOOKS;
+  }
+
+  @Override
+  public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, statKey, maxLevel, amount, modifier, allowBroken, heldTag);
+      });
+
```

Comment:
Why are you fetching the capability here if you never use `data`? You also get it inside `addStatIfArmor`. Just call the `addStatIfArmor` helper directly.